test/tools/evmoneRunner.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/tools/evmoneRunner.h>
+#include <iostream>
+#include <assert.h>
+#include <fstream>
+#include <limits>
+
+using namespace external::evmone;
+using namespace std;
+
+evmc_host_interface EvmOneVM::interface{
+		{},
+		{},
+		{},
+		{},
+		{},
+		{},
+		{},
+		{},
+		{},
+		[](evmc_context* ctx) { return static_cast<EvmOneVM*>(ctx)->tx_context; },
+		{},
+		{},
+};
+
+void EvmOneVM::execute(const evmc_message& _m, std::basic_string<uint8_t> _code)
+{
+	evmc_result result = {};
+	result = vm->execute(vm, this, rev, &_m, &_code[0], _code.size());
+	assert(result.status_code == EVMC_SUCCESS);
+	// Checking if execution output is true or 1
+	//	assert(result.output_data[31] == '\x01');
+}
+
+void EvmOneVM::execute(int64_t gas, std::basic_string<uint8_t> _code, std::string _hexInput)
+{
+	auto input = from_hex(_hexInput);
+	evmc_message msg = {};
+	msg.gas = gas;
+	msg.input_data = input.data();
+	msg.input_size = input.size();
+	execute(msg, _code);
+}
+
+void EvmOneVM::execute(std::basic_string<uint8_t> _code, std::string _hexInput)
+{
+	execute(std::numeric_limits<int64_t>::max(), _code, _hexInput);
+}
+
+void EvmOneVM::execute(int64_t _gas, std::string _code, std::string _input)
+{
+	execute(_gas, from_hex(_code), _input);
+}
+
+void EvmOneVM::execute(std::string _code, std::string _hexInput)
+{
+	execute(std::numeric_limits<int64_t>::max(), _code, _hexInput);
+}
+
+// Converts hex encoded string to its raw byte representation
+std::basic_string<uint8_t> EvmOneVM::from_hex(std::string _hex)
```

Comment:
`dev::asString(dev::fromHex(_hex))`

Reply:
If you only need a raw pointer, you don't even need `asString`.

Reply:
`dev::fromHex(_hex)` returns value of type `std::vector<unsigned char>` but evmone api needs value of type `std::basic_string<unsigned char>`

Reply:
evmone uses EVMC API which passes buffers as two arguments: pointer + size. No fancy types are needed. https://github.com/ethereum/solidity/pull/6838/files#diff-b845a4b0200f220649c1ef84ed2ba2c7R61

Reply:
sure, but I still need this utility function, right? https://github.com/ethereum/solidity/pull/6838/files#diff-b845a4b0200f220649c1ef84ed2ba2c7R82